test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs

```diff
@@ -0,0 +1,488 @@
+                }
+                else
+                {
+                    Assert.DoesNotContain("\"@odata.type\":\"#Microsoft.OData.E2E.TestCommon.Common.Server.EndToEnd.ContactDetails\"", result);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Provide EDM model, and then write an entry with wrong serialization info
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task WriteEntryWithWrongSerializationInfoWithModel(string mimeType)
+        {
+            bool[] autoComputeMetadataBools = new bool[] { true, false, };
+
+            foreach (var autoComputeMetadata in autoComputeMetadataBools)
+            {
+                var settings = new ODataMessageWriterSettings();
+                settings.ODataUri = new ODataUri() { ServiceRoot = _baseUri };
+
+                var responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+                responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+
+                var personType = _model.FindDeclaredType(NameSpacePrefix + "Person") as IEdmEntityType;
+                var peopleSet = _model.EntityContainer.FindEntitySet("People");
+
+                using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings, _model))
+                {
+                    var odataWriter = await messageWriter.CreateODataResourceWriterAsync(peopleSet, personType);
+                    var entry = this.CreatePersonEntryWithoutSerializationInfo();
+
+                    entry.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "Parsen", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+                    await odataWriter.WriteStartAsync(entry);
+                    await odataWriter.WriteEndAsync();
+                    var result = await TestsHelper.ReadStreamContentAsync(responseMessageWithoutModel.GetStream());
+                    Assert.Contains("\"PersonId\":-5", result);
+                    if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                    {
+                        // no metadata does not write odata.metadata
+                        Assert.Contains("$metadata#Parsen/$entity", result);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Change serialization info value after WriteStart
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task ChangeSerializationInfoAfterWriteStart(string mimeType)

```

Comment:
What is the expected outcome of this scenario?

Reply:
updated